src/huggingface_hub/hf_api.py

```diff
@@ -6926,6 +6926,49 @@ def _get_namespace(self, token: Optional[str] = None) -> str:
     ########################
     # Collection Endpoints #
     ########################
+    @validate_hf_hub_args
+    def list_collections(
+        self,
+        *,
+        owner: Union[List[str], str, None] = None,
+        item: Union[List[str], str, None] = None,
+        sort: Optional[Literal["last_modified", "trending", "upvotes"]] = None,
+        limit: Optional[int] = None,
+        token: Optional[Union[bool, str]] = None,
+    ) -> Iterable[Collection]:
+        """List collections on the Huggingface Hub, given some filters.
+
+        Args:
+            owner (`List[str]` or `str`, *optional*):
+                Filter by owner's username.
+            item (`List[str]` or `str`, *optional*):
+                Select particular items to return.
```

Comment:
```suggestion
                Filter collections containing a particular items. Example: `"models/teknium/OpenHermes-2.5-Mistral-7B"`, `"datasets/squad"` or `"papers/2311.12983"`.
```

(nit) I like the example you listed in the docs.